src/atc/configurator/configurator.py

```diff
@@ -39,57 +52,103 @@ def __init__(self, resource_path: Union[str, ModuleType] = None):
 
     def clear_all_configurations(self):
         self._raw_resource_details = dict()
-        self._extra_config = dict()
         self._is_debug = False
         self.table_details = dict()
+        self._set_extras()
 
     ############################################
     # the core logic of this class is contained
     # in the following methods
     ############################################
 
+    def _set_extras(self):
+        self.register("ID", {"release": "", "debug": f"__{self._unique_id}"})
+        self.register("MNT", {"release": "mnt", "debug": "tmp"})
+
+    @deprecation.deprecated(
+        details="register literal string values instead.",
+    )
     def get_extra_details(self) -> Dict[str, str]:
-        """get all special substitutions not based on resources"""
-        extras = {
-            "ID": f"__{self._unique_id}" if self._is_debug else "",
-            "MNT": "tmp" if self._is_debug else "mnt",
-        }
-        extras.update(self._extra_config)
-        return extras
-
-    def _get_item(self, table_id: str) -> Dict[str, str]:
+        """The distinction between extras and normal values was removed.
+        Simply register the value using the .register method"""
+        return {}
```

Comment:
Getting the extras was previously used internally. I see no reason why anyone should have used this before. Internally it is not needed any more. The internals do not distinguish between extras and registrations any more, so I don't have an easy way to get them back to the user here, and I also don't find it very important.

Reply:
Should the method be removed then? Why would anybody use get_extra_details() if it always return an empty string? 

```python
def get_extra_details(self) -> Dict[str, str]:
        """The distinction between extras and normal values was removed.
        Simply register the value using the .register method"""
        return {}
```

Reply:
I left it in as deprecated because I don't want to remove any methods in this refactoring. It is not intended to be used by anyone. It already has the deprecated decorator. I will extend the comments with more details.